### PR TITLE
GDK Tools dialog auto-saves

### PR DESCRIPTION
**Pull request by austinkelmore**
#### Description
Made the GDK Tools dialog auto-save. Also moved the reset button to the bottom instead of the top to make it a bit cleaner.

#### Documentation
* [x] Changelog
* [x] Documentation

### CODE REVIEW

**Comment by austinkelmore** [comment]
> LGTM, though I'm not entirely sure about that timer. Not sure what could replace it though.

I'm not a huge fan of the timer, either. I wish Unity had a concept of "finished entering in text, go ahead and update whatever you need to". I've copied the timer from other sections of code in the GDK, so it's at least consistent.

File: workers/unity/Packages/io.improbable.gdk.tools/GdkToolsConfigurationWindow.cs [hunk 1]
```diff
@@ -224,5 +248,22 @@ private void DrawCustomSnapshotDir()
...
+        {
+            var timeSinceLastSave = DateTime.Now - lastSaveTime;
+            if (!hasUnsavedData || timeSinceLastSave < FileSavingInterval)
+            {
+                return;
+            }
+
+            toolsConfig.Save();
+            lastSaveTime = DateTime.Now;
+        }
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.tools/GdkToolsConfigurationWindow.cs:267]: I think there's a race here where we could save invalid data.

1. User inputs some valid value, `hasUnsavedData` is set to true.
2. User changes a value to invalid. 
3. The interval passes and we are able to save.
4. Invalid data is saved

There should be a check against any config errors before we are able to save: 

```csharp
if (!hasUnsavedData || timeSinceLastSave < FileSavingInterval || configErrors.Any())
```

File: workers/unity/Packages/io.improbable.gdk.tools/GdkToolsConfigurationWindow.cs [hunk 2]
```diff
@@ -224,5 +248,22 @@ private void DrawCustomSnapshotDir()
...
+        private void TrySaveChanges()
+        {
+            var timeSinceLastSave = DateTime.Now - lastSaveTime;
+            if (!hasUnsavedData || timeSinceLastSave < FileSavingInterval)
+            {
+                return;
+            }
+
+            toolsConfig.Save();
+            lastSaveTime = DateTime.Now;
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.tools/GdkToolsConfigurationWindow.cs:266]: Shouldn't we set `hasUnsavedData` to `false` after saving?

Comment (austinkelmore) [reply workers/unity/Packages/io.improbable.gdk.tools/GdkToolsConfigurationWindow.cs:266]: Oh goodness. Good catch!

File: workers/unity/Packages/io.improbable.gdk.tools/GdkToolsConfigurationWindow.cs (hunk 1 above)
Comment (austinkelmore) [reply workers/unity/Packages/io.improbable.gdk.tools/GdkToolsConfigurationWindow.cs:267]: Sounds good, making this change.

**Pull request merged**